experimental/Schemes/singular_locus.jl

```diff
@@ -3,45 +3,63 @@ export singular_locus
 export reduced_scheme
 export is_smooth
 
-function singular_locus(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
-  comp = _singular_locus_with_decomposition(X)
+QuoRing = Union{MPolyQuoLocalizedRing, 
+                MPolyQuo
+               }
+
+### TODO: The following two functions need to be made type-sensitive 
+###       and reduced=true needs to be set for varieties 
+###       as soon as not only schemes, but also varieties as special 
+###       schemes have been introduced
+function singular_locus(X::AbsSpec{<:Ring, <:QuoRing})
+  comp = _singular_locus_with_decomposition(X,reduced=false)
   if length(comp) == 0 
     return subscheme(X, one(OO(X)))
   end
   R = base_ring(OO(X))
-  return Spec(R, prod([modulus(quotient_ring(OO(Y))) for Y in comp]), inverted_set(OO(X)))
+  I= prod([modulus(quotient_ring(OO(Y))) for Y in comp])
+  return Spec(R, I, inverted_set(OO(X)))
 end
 
-function _singular_locus_with_decomposition(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
-  I = modulus(OO(X))
+function singular_locus_reduced(X::AbsSpec{<:Ring, <:QuoRing})
+  comp =  _singular_locus_with_decomposition(X, reduced=true)
+  I= ideal(ambient_ring(X),one(ambient_ring(X)))
+  for Z in comp
+    I = intersect(I, modulus(quotient_ring(Z)))
+  end     
+  return subscheme(X,I)
+end
+
+function _singular_locus_with_decomposition(X::AbsSpec{<:Ring, <:QuoRing}, reduced::Bool=true)
+  I = saturated_ideal(modulus(OO(X)))
```

Comment:
This will probably crash when the input is an `MPolyQuo`.

Reply:
Wait until the next git push tonight. Then you see, where I am heading with this and how this point is solved.

Reply:
Alright! I just wanted to give you a heads up -- just in case...